server/storage/memory.go

```diff
@@ -158,6 +181,97 @@ func (st *MemStorage) Delete(gun string) error {
 	return nil
 }
 
+// GetChanges returns a []Change starting from but excluding the record
+// identified by changeID. In the context of the memory store, changeID
+// is simply an index into st.changes. The ID of a change is its
+// index+1, both to match the SQL implementations, and so that the first
+// change can be retrieved by providing ID 0.
+func (st *MemStorage) GetChanges(changeID string, pageSize int, filterName string, reversed bool) ([]Change, error) {
+	id, err := strconv.ParseInt(changeID, 10, 32)
+	if err != nil {
+		return nil, err
+	}
+	var (
+		start     = int(id)
+		toInspect []Change
+	)
+	if err != nil {
+		return nil, err
+	}
+	if len(st.changes) <= int(id) && !reversed {
+		// no records to return as we're essentially trying to retrieve
+		// changes that haven't happened yet.
+		return nil, nil
```

Comment:
I had that originally but I decided it should be OK to request a range that doesn't exist with the expected result being an empty list. I created `ErrBadChangeID` to indicate the ID is unrecognizable as an ID.